src/JsonRenderer.cpp

```diff
@@ -118,40 +121,24 @@ unsigned JsonRenderFloat(double n, char* pBuffer) throw()
     return cch;
 }
 
-unsigned JsonRenderBool(bool b, char* pBuffer) throw()
+unsigned JsonRenderBool(bool b, _Out_writes_z_(6) char* pBuffer) throw()
 {
-    unsigned cch;
-    if (b)
-    {
-        strcpy(pBuffer, "true");
-        cch = 4;
-    }
-    else
-    {
-        strcpy(pBuffer, "false");
-        cch = 5;
-    }
-    return cch;
+    return b ? MemCpyFromLiteral(pBuffer, "true") : MemCpyFromLiteral(pBuffer, "false");
 }
 
-unsigned JsonRenderNull(char* pBuffer) throw()
+unsigned JsonRenderNull(_Out_writes_z_(5) char* pBuffer) throw()
 {
-    strcpy(pBuffer, "null");
-    return 4;
+    return MemCpyFromLiteral(pBuffer, "null");
 }
 
-unsigned JsonRenderTime(
-    std::chrono::system_clock::time_point const& timePoint,
-    char* pBuffer) throw()
+static unsigned RenderTicks1970(time_t seconds1970, unsigned subsecondTicks, _Out_writes_z_(29) char* pBuffer) throw()
 {
-    time_t printableTime = std::chrono::system_clock::to_time_t(timePoint);
-    tm timeStruct = tm();
-    gmtime_r(&printableTime, &timeStruct);
-
-    auto subsecondDuration =
-        timePoint.time_since_epoch() % std::chrono::seconds{ 1 };
-    auto subsecondNanos =
-        std::chrono::duration_cast<std::chrono::nanoseconds>(subsecondDuration);
+    tm timeStruct = {};
+#ifdef _WIN32
+    _gmtime64_s(&timeStruct, &seconds1970);
```

Comment:
Note: TODO for a subsequent PR - Windows implementation of `_gmtime64_s` has limited range. Probably better to implement this using FileTimeToSystemTime.